apps/participants/utils.py

```diff
@@ -0,0 +1,28 @@
+from challenges.models import Challenge
+
+from .models import Participant
+
+
+def is_user_part_of_participant_team(user, participant_team_id):
+    """Returns boolean if the user belongs to the participant team or not"""
+    return Participant.objects.filter(user=user, team__id=participant_team_id).exists()
+
+
+def has_participant_team_participated_in_challenge(participant_team_id, challenge_id):
+    """Returns boolean if the Participant Team participated in particular Challenge"""
+    return Challenge.objects.filter(pk=challenge_id, participant_team__id=participant_team_id).exists()
+
+
+def has_user_participated_in_challenge(user, challenge_id):
+    """Returns boolean if the user has participated in a particular challenge"""
+    participant_teams = Participant.objects.filter(user=user).values_list('teams', flat=True)
+    return Challenge.objects.filter(pk=challenge_id, participant_teams__in=participant_teams).exists()
+
+
+def get_participant_team_id_of_user_for_a_challenge(user, challenge_id):
```

Comment:
I am pretty sure that there is a better way of doing this. Will look into it. For now, I think we are good with this.